### PR TITLE
Fixed selector for DaemonSet

### PR DESCRIPTION
**Pull request by soltysh**
DaemonSet uses new labels, similarly to jobs, see [here](https://github.com/kubernetes/kubernetes/blob/master/docs/user-guide/jobs.md#pod-selector). I'll submit a PR to fix the docs in k8s regarding DS selector, which are currently wrong.

### CODE REVIEW

File: deployment/templates/fluentd.yaml [hunk 1]
```diff
@@ -31,8 +31,9 @@ objects:
...
       replicas: 0
       selector:
-        provider: openshift
-        component: "fluentd"
+        matchLabels:
+          provider: openshift
+          component: "fluentd"
       triggers:
       - type: ConfigChange
       - type: ImageChange
```
Comment (0xmichalis) [inline deployment/templates/fluentd.yaml]: Do DaemonSets support image change triggers? Or any kind of trigger for that matter?

Comment (soltysh) [inline deployment/templates/fluentd.yaml]: Good point - updating.

**Comment by soltysh** [comment]
Removed triggers and replaced with analogical rolling strategy for daemon set.

**Comment by ewolinetz** [comment]
Thanks @soltysh !

**Pull request merged**